Added specific rendering for shop=bed.

hikemaniac opened the pull request:
This PR adds a special icon for shop=bed. As this is my first time contributing to this repository, please verify the correctness of my commits structure.

This is what the icon looks like on 14x14:
![bed](https://user-images.githubusercontent.com/31667811/37355982-465a768e-26e5-11e8-84b5-abd5fce01137.png)
This PR closes #3118.

hikemaniac commented:
Just to be sure that this is the way you meant it:
![bed-inverted](https://user-images.githubusercontent.com/31667811/37359865-f7bcb0dc-26ee-11e8-953a-8cbf6406ec70.png)
It takes a bit to get used to, but in general I think that this solution is a good one.

kocio-pl commented:
Yes, it was. But I still need to see the real map rendering with this element - do you have the testing environment ready?

lakedistrictOSM commented:
Could you try making the headboard a little taller?

hikemaniac commented:
@kocio-pl Because of the test rendering: I successfully installed Docker, added a data.osm.pbf file and ran the line `docker-compose up import`. When running the command `docker-compose up kosmtik` I recieved an error after some time.

![kosmtik installation failing](https://user-images.githubusercontent.com/31667811/37539078-f839bd64-2952-11e8-85e9-60f2442e4792.png)

I retried it and again the same error message appeared. Can anyone help?
I'm using **Docker Community Edition 17.12.0** and **macOS 10.10.5**.

kocio-pl commented:
I would try with rebuilding Docker image forcing `apt-get update` (like adding a line before it) or just skipping cache entirely:

https://stackoverflow.com/questions/35594987/how-to-force-docker-for-clean-build-of-an-image#35595021

However this is not about lacking packages - but maybe problem with retrieving proper hashes? Another option would be to log into the running docker container and testing the files inside - probably like a root:

`sudo docker exec -i -t -u root loving_heisenberg /bin/bash`

hikemaniac commented:
@kocio-pl I'm really sorry, but I couldn't generate an example rendering. It is still exactly the same problem. I tried installing Kosmtik with no cache, but Docker complained about that (maybe I messed up the syntax?), I tried to use your command and I even updated my OS because Docker also claimed that my macOS version could eventually lead to problems, but nothing helped. I've run out of ideas now and so I would be really thankful if you could make a test rendering.

kocio-pl commented:
Well, the basic idea left to check is to run VirtualBox, which is available for popular systems, and install Ubuntu (64-bit) inside, then run Docker inside Ubuntu. When just testing some simple design, performance is not the top priority.

kocio-pl commented:
Could you insert this icon: 

https://gist.github.com/gpsvisualizer/320b7898ee88e5a9a99d80890ee3f831

and fix merging conflicts? I would then make the tests myself, OK?

hikemaniac commented:
@kocio-pl I just replaced the icon by @gpsvisualizer's icon. I really like the new one. 👍 I also fixed the conflicts with master. Sorry for not having the time to set up a virtual machine for Docker, but I think this might have exceeded my computers spare memory. :disappointed_relieved:

kocio-pl commented:
Regarding Docker problems - I have found something worth testing (adding to the Dockerfiles):

https://github.com/moby/moby/issues/23203#issuecomment-223328829

kocio-pl commented:
Could you update the icon shape to https://gist.github.com/gpsvisualizer/8817bd6b3ad0b037d8b2e1e9a928bd3c? It's very close, but just looks better because of the stronger lines:

[Example shop](https://www.openstreetmap.org/node/5072352966):

Current icon
![g8qoys1o](https://user-images.githubusercontent.com/5439713/38159305-dcb5732c-34a5-11e8-8ff9-b9322d446c42.png)

Updated icon
![pxovcznp](https://user-images.githubusercontent.com/5439713/38159308-e127108c-34a5-11e8-9eaf-eedf7fbffee9.png)

hikemaniac commented:
The test rendering looks great! I replaced the icon with the version with the thicker lines.

kocio-pl commented on amenity-points.mss:
```diff
@@ -743,6 +743,10 @@
     [shop = 'beauty'][zoom >= 18] {
       marker-file: url('symbols/shop/beauty.svg');
     }
+    
```
Unwanted spaces, please remove them.

kocio-pl commented:
One more technical detail - could you squash all the commits into one with clear name?

kocio-pl commented:
There are now 9 commits, did you try squashing them?

kocio-pl commented:
There's a conflict in`project.mml` now, please resolve it.

hikemaniac commented:
@kocio-pl I don't have any luck with this PR. I tried sqaushing my commits and in return git drowned me in a bunch of merging conflicts (no idea what I'm doing wrong). I then used a GUI to try squashing them, but it told me that I can't squash merge commits (all my "rebasing-and-resolving-conflicts-commits). However there might be a way to still squash them. I assume that there is a merging option called "squash and merge" shown to you in the green merging dropdown on GitHub. I have never actually tried it but it should squash all commits into a single one and merge them into master. I really hope it works. 👍

matthijsmelissen commented:
@kocio-pl Squashing can also be done at mergetime by the merger. I think it is preferable (because the PR author never knows when his PR is done).

kocio-pl commented on project.mml:
```diff
@@ -1987,7 +1987,8 @@ Layer:
               'advertising_' || CASE WHEN tags->'advertising' in ('column') THEN tags->'advertising' else NULL END,
-              'shop_' || CASE WHEN shop IN ('supermarket', 'bag', 'bakery', 'beauty', 'books', 'butcher', 'clothes', 'computer', 'confectionery',
+              'shop_' || CASE WHEN shop IN ('supermarket', 'bag', 'bakery', 'beauty', 'bed', 'books', 'butcher', 'clothes', 'computer', 'confectionery',
+
```
This empty line should be removed.

hikemaniac commented:
I removed the unneccesary line.

lakedistrictOSM commented:
Thanks @gpsvisualizer and @hikemaniac for your work on this! 😃

jbelien commented on project.mml:
```diff
@@ -2132,9 +2132,11 @@ Layer:
                                                       'car_wash', 'post_box', 'post_office', 'recycling', 'telephone', 'toilets', 'taxi',
                                                       'drinking_water', 'hunting_stand', 'nightclub', 'veterinary', 'social_facility',
+                                                      'charging_station', 'arts_centre', 'ferry_terminal', 'marketplace') THEN amenity ELSE NULL END,
+                  'shop_' || CASE WHEN shop IN ('supermarket', 'bag','bakery', 'beauty', 'bed', 'books', 'butcher', 'clothes', 'computer', 'confectionery', 'fashion',
```
Probably a stupid question, but I just want to understand how this work : why shop_ is "duplicated" on lines 2136 and 2139 ? Thanks :)

jbelien replied:
I looked a bit more in details and I think the 2136 line should NOT be here.

I think it should be :
```
'amenity_' || CASE WHEN amenity IN ('pub', 'restaurant', 'food_court', 'cafe', 'fast_food', 'biergarten', 'bar', 'library', 'theatre',
                                  'courthouse', 'townhall', 'cinema', 'clinic', 'community_centre', 'parking', 'bicycle_parking', 'public_bath',
                                  'motorcycle_parking', 'police', 'fire_station', 'fountain', 'place_of_worship', 'grave_yard', 'shelter', 'bank',
                                  'embassy', 'fuel', 'bus_station', 'prison', 'university', 'school', 'college', 'kindergarten', 'hospital',
                                  'ice_cream', 'pharmacy', 'doctors', 'dentist', 'atm', 'bicycle_rental', 'car_rental',
                                  'car_wash', 'post_box', 'post_office', 'recycling', 'telephone', 'toilets', 'taxi',
                                  'drinking_water', 'hunting_stand', 'nightclub', 'veterinary', 'social_facility',
                                  'charging_station', 'arts_centre', 'ferry_terminal', 'marketplace', 'shower', 'bbq') THEN amenity ELSE NULL END,
'advertising_' || CASE WHEN tags->'advertising' in ('column') THEN tags->'advertising' else NULL END,
'shop_' || CASE WHEN shop IN ('supermarket', 'bag', 'bakery', 'beauty', 'bed', 'books', 'butcher', 'clothes', 'computer', 'confectionery', 'fashion',
                            'convenience', 'department_store', 'doityourself', 'hardware', 'fishmonger', 'florist', 'garden_centre', 'hairdresser',
                            'hifi', 'ice_cream', 'car', 'car_repair', 'bicycle', 'mall', 'pet', 'photo', 'photo_studio', 'photography',
                            'seafood', 'shoes', 'alcohol', 'gift', 'furniture', 'kiosk', 'mobile_phone', 'motorcycle', 'musical_instrument',
                            'newsagent', 'optician', 'jewelry', 'jewellery', 'electronics', 'chemist', 'toys', 'travel_agency', 'car_parts',
                            'greengrocer', 'farm', 'stationery', 'laundry', 'dry_cleaning', 'beverages', 'perfumery', 'cosmetics',
                            'variety_store', 'wine', 'outdoor', 'copyshop', 'sports', 'deli', 'tobacco', 'art', 'tea', 'coffee', 'tyres',
                            'pastry', 'chocolate', 'music', 'medical_supply','dairy', 'video_games') THEN shop
              WHEN shop IN ('no', 'vacant', 'closed', 'disused', 'empty') OR shop IS NULL THEN NULL ELSE 'other' END,
```
instead of :
```
'amenity_' || CASE WHEN amenity IN ('pub', 'restaurant', 'food_court', 'cafe', 'fast_food', 'biergarten', 'bar', 'library', 'theatre',
                                  'courthouse', 'townhall', 'cinema', 'clinic', 'community_centre', 'parking', 'bicycle_parking', 'public_bath',
                                  'motorcycle_parking', 'police', 'fire_station', 'fountain', 'place_of_worship', 'grave_yard', 'shelter', 'bank',
                                  'embassy', 'fuel', 'bus_station', 'prison', 'university', 'school', 'college', 'kindergarten', 'hospital',
                                  'ice_cream', 'pharmacy', 'doctors', 'dentist', 'atm', 'bicycle_rental', 'car_rental',
                                  'car_wash', 'post_box', 'post_office', 'recycling', 'telephone', 'toilets', 'taxi',
                                  'drinking_water', 'hunting_stand', 'nightclub', 'veterinary', 'social_facility',
                                  'charging_station', 'arts_centre', 'ferry_terminal', 'marketplace') THEN amenity ELSE NULL END,
'shop_' || CASE WHEN shop IN ('supermarket', 'bag', 'bakery', 'beauty', 'bed', 'books', 'butcher', 'clothes', 'computer', 'confectionery', 'fashion',
                                  'charging_station', 'arts_centre', 'ferry_terminal', 'marketplace', 'shower', 'bbq') THEN amenity ELSE NULL END,
'advertising_' || CASE WHEN tags->'advertising' in ('column') THEN tags->'advertising' else NULL END,
'shop_' || CASE WHEN shop IN ('supermarket', 'bag', 'bakery', 'beauty', 'bed', 'books', 'butcher', 'clothes', 'computer', 'confectionery', 'fashion',
                            'convenience', 'department_store', 'doityourself', 'hardware', 'fishmonger', 'florist', 'garden_centre', 'hairdresser',
                            'hifi', 'ice_cream', 'car', 'car_repair', 'bicycle', 'mall', 'pet', 'photo', 'photo_studio', 'photography',
                            'seafood', 'shoes', 'alcohol', 'gift', 'furniture', 'kiosk', 'mobile_phone', 'motorcycle', 'musical_instrument',
                            'newsagent', 'optician', 'jewelry', 'jewellery', 'electronics', 'chemist', 'toys', 'travel_agency', 'car_parts',
                            'greengrocer', 'farm', 'stationery', 'laundry', 'dry_cleaning', 'beverages', 'perfumery', 'cosmetics',
                            'variety_store', 'wine', 'outdoor', 'copyshop', 'sports', 'deli', 'tobacco', 'art', 'tea', 'coffee', 'tyres',
                            'pastry', 'chocolate', 'music', 'medical_supply','dairy', 'video_games') THEN shop
              WHEN shop IN ('no', 'vacant', 'closed', 'disused', 'empty') OR shop IS NULL THEN NULL ELSE 'other' END,
```

hikemaniac replied:
Thanks for catching and fixing this error. This probably happened when resolving merging conflicts. 😅

kocio-pl replied:
I wish we had [separate whitelists](https://github.com/gravitystorm/openstreetmap-carto/issues/2452) instead of this spaghetti SQL code...
